File: 59e95a4689eeb92f380f4ab2_202007_5b88a998-d225-11ea-9fea-00163e169c3a_源代码/代码/第3章/2_Memeory/2_Memeory/ViewController.m
//
//  ViewController.m
//  2_Memeory
//
//  Created by jaki on 2019/9/20.
//  Copyright © 2019 jaki. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    NSLog(@"%p", @"Hello");
    
    static int data = 1;
    static int bss;
    static int data2 = 1;
    static int bss2;
    NSLog(@"data:%p, bss:%p", &data, &bss);
    NSLog(@"data2:%p, bss2:%p", &data2, &bss2);
    
    NSObject *object = [[NSObject alloc] init];
    NSObject *object2 = [[NSObject alloc] init];
    NSLog(@"%p, %p", object, &object);
    NSLog(@"%p, %p", object2, &object2);
    NSLog(@"%d", sizeof(NSObject *));
}


@end

File: 59e95a4689eeb92f380f4ab2_202007_5b88a998-d225-11ea-9fea-00163e169c3a_源代码/代码/第6章/Capacity/Capacity/ViewController.swift
//
//  ViewController.swift
//  Capacity
//
//  Created by 珲少 on 2020/1/14.
//  Copyright © 2020 jaki. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
    
    class ViewControllerCell: UITableViewCell {
        override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
            super.init(style: style, reuseIdentifier: reuseIdentifier)
            self.setupUI()
        }
        
        required init?(coder: NSCoder) {
            fatalError("init(coder:) has not been implemented")
        }
        
        func setupUI() {
            for i in 0 ... 2000 {
                let v = UIView(frame: CGRect(x:0, y: 0, width: 10, height: 10))
                v.backgroundColor = UIColor.green
                v.isOpaque = true
                self.contentView.addSubview(v)
            }
        }
    }
    
    
    lazy var tableView:UITableView! = {
        let tableView = UITableView(frame: self.view.bounds, style: .plain)
        tableView.register(ViewControllerCell.classForCoder(), forCellReuseIdentifier: "ViewControllerCell")
        tableView.delegate = self
        tableView.dataSource = self
        return tableView
    }()

    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.view.addSubview(self.tableView)
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 1000
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "ViewControllerCell", for: indexPath)
        return cell
    }
    
}

